csharp/src/Ice/Reference.cs

```diff
@@ -48,6 +48,8 @@ public sealed class Reference : IEquatable<Reference>
         internal Protocol Protocol { get; }
         internal RouterInfo? RouterInfo { get; }
         private int _hashCode;
+        private IReadOnlyList<InvocationInterceptor> _invocationInterceptors;
```

Comment:
_invocationInterceptors should be readonly. Also, why not make them accessible to the application like Location/LocatorInfo etc., as IObjectPrx.InvocationInterceptors ?

Reply:
added readonly modifier, if we make this accessible we should do the same with the communicator and adapter interceptors